src/trace/context/extractors/app-sync.ts

```diff
@@ -0,0 +1,18 @@
+import { HTTPEventTraceExtractor } from "./http";
+import { EventTraceExtractor } from "../extractor";
+import { TracerWrapper } from "../../tracer-wrapper";
+import { SpanContextWrapper } from "../../span-context-wrapper";
+
+export class AppSyncEventTraceExtractor implements EventTraceExtractor {
+  private httpEventExtractor: HTTPEventTraceExtractor;
+
+  constructor(private tracerWrapper: TracerWrapper) {
+    this.httpEventExtractor = new HTTPEventTraceExtractor(this.tracerWrapper, false);
+  }
+
+  extract(event: any): SpanContextWrapper | null {
+    event.headers = event.request.headers;
```

Comment:
Yes, that is correct. It mutates the object – as far as I could tell from unit testing.

I think this is not ideal, but this is what the previous code, before refactoring, did. I imagine we should update this in the future. Either way, a more thorough test should be done here!

Reply:
What if instead of using a `this.httpEventExtractor` you just use a regular extractor? Instead of passing an event object with a `headers` attr, you just pass the headers themselves?

Reply:
I'm not sure how AppSync events work, but I wonder if they could contain authorizer context – or headers with mixed casing, thus the original author using this approach – so, I agree with @astuyve on just moving `app-sync` into the http event extractor 🤔  Wdyt @purple4reina? Also, what do you mean by a "regular extractor"?

Reply:
@duncanista okay apologies, I took a closer look at this and retract my comment about "regular extractor". I did not realize we were doing more with the event than just passing `event.headers` to the extractor.

I support moving `app-sync` into the http event extractor. 👍🏽

Reply:
Will do in another PR – to avoid making multiple non-w3c related changes here.